src/components/general/cart-item/index.tsx

```diff
@@ -0,0 +1,88 @@
+import styled from 'styled-components';
+import { Button } from '../button';
+import { IconButton } from '../icon-button/IconButton';
+
+type alterAmount = 'increase' | 'decrease';
+
+type CartItemProp = {
+  imgSrc: string;
+  name: string;
+  quantity: number;
+  alterQuantity: (alter: alterAmount) => void;
```

Comment:
I see what you're trying to do with this but imo it would be better to give this component a _really_ straight forward interface with functions like `onRemoveClick`, `onIncreaseStepperClick` and `onDecreaseStepperClick`. 

Making abstractions like `alterQuantity` is tempting but in the wild your PM could ask for some analytics to be attached to when customers are decreasing quantity and a confetti effect to be shown when customers increase quantity, at that point the early refactor becomes a roadblock

Reply:
Ah yeah that makes sense, will have a rework